### PR TITLE
the line number of the yaml file is displayed in the error message

### PR DESCRIPTION
**Pull request by aalbino2**

### CODE REVIEW

**Comment by aalbino2** [comment]
I also added the support to "extends" in the yaml file, as aske by @domna 

Note that I also changed the file used for testing:
/nomad/dependencies/parsers/nexus/tests/data/tools/dataconverter/readers/mpes/Ref_nexus_mpes.log

Because some field was changed in the nxdl from recommended or required to optional.

**Review by domna — APPROVED** [review]
The extends keyword works like a charm. Did not review anything else.

The Ref log output for mpes should be the same as in the main branch now, because I updated it only yesterday.

**Review by sanbrock — CHANGES_REQUESTED** [review]
it would be great to have some test cases where the new feature is checked

File: nexusparser/tools/yaml2nxdl/yaml2nxdl.py [hunk 1]
```diff
@@ -108,7 +114,10 @@ def yaml2nxdl(input_file: str, verbose: bool):
         del yml_appdef['category']
```
Comment (shabihsherjeel) [inline nexusparser/tools/yaml2nxdl/yaml2nxdl.py:114]: This line and line no: 111 can be removed out of the two if statements. It is always called  anyways.

File: nexusparser/tools/yaml2nxdl/yaml2nxdl_forward_tools.py [hunk 2]
```diff
@@ -387,38 +480,41 @@ def recursive_build(obj, dct, verbose):
...
-        elif keyword_name != '':
-            fld = not_empty_keyword_name_handle(obj, keyword_type, keyword_name)
-            second_nested_level_handle(verbose, fld, value)
-            second_level_attributes_handle(fld, keyword, value)
+        elif keyword_name != '' and '__line__' not in keyword_name:
+            fld = not_empty_keyword_name_handle(
+                obj, keyword_type, keyword_name)
+            second_nested_level_handle(verbose, value, fld)
+            second_level_attributes_handle(dct, fld, keyword, value)
         else:
```
Comment (shabihsherjeel) [inline nexusparser/tools/yaml2nxdl/yaml2nxdl_forward_tools.py:519]: This else could be removed.

**Review by shabihsherjeel — CHANGES_REQUESTED** [review]

**Review by shabihsherjeel — APPROVED** [review]

**Review by sanbrock — APPROVED** [review]

**Pull request merged**